### PR TITLE
Fix contract

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Contracts build ok

### CODE REVIEW

File: crml/erc20-peg/contracts/ERC20Peg.sol [hunk 1]
```diff
@@ -28,7 +28,7 @@ contract ERC20Peg is Ownable {
         require(depositsActive, "deposits paused");
 
         // CENNZ deposits will require a vote to activate
-        if (address == 0x1122b6a0e00dce0563082b6e2953f3a943855c1f) {
+        if (address(0) == 0x1122B6a0E00DCe0563082b6e2953f3A943855c1F) {
```
Comment (KarishmaBothara) [inline crml/erc20-peg/contracts/ERC20Peg.sol:31]: used https://ethsum.netlify.app/ to get correct checksum

File: crml/erc20-peg/contracts/ERC20Peg.sol [hunk 2]
```diff
@@ -43,9 +43,8 @@ contract ERC20Peg is Ownable {
     // i.e. v[i], r[i], s[i] matches the i-th validator[i]
     function withdraw(address tokenType, uint256 amount, uint256 event_id, uint8[] memory v, bytes32[] memory r, bytes32[] memory s) payable external {
         require(withdrawalsActive, "withdrawals paused");
-        bytes32 message = abi.encodePacked(tokenType, amount, msg.sender);
-        bridge.call(bytes4(keccak256("verifyMessage(bytes,uint256,uint8[],bytes32[],bytes32[])")), message, event_id, v, r, s); 
-
+        bytes32 message = keccak256(abi.encodePacked(tokenType, amount, msg.sender));
+        bridge.call(abi.encodeWithSignature("verifyMessage(bytes,uint256,uint8[],bytes32[],bytes32[])", message, event_id, v, r, s));
```
Comment (KarishmaBothara) [inline crml/erc20-peg/contracts/ERC20Peg.sol:47]: as per solidity docs (breaking changes)
https://docs.soliditylang.org/en/latest/050-breaking-changes.html?highlight=keccak

File: crml/erc20-peg/contracts/ERC20Peg.sol (hunk 1 above)
Comment (jordy25519) [reply crml/erc20-peg/contracts/ERC20Peg.sol:31]:
```suggestion
        if (address(tokenType) == 0x1122B6a0E00DCe0563082b6e2953f3A943855c1F) {
```

File: crml/erc20-peg/contracts/ERC20Peg.sol [hunk 3]
```diff
@@ -43,9 +43,8 @@ contract ERC20Peg is Ownable {
     // i.e. v[i], r[i], s[i] matches the i-th validator[i]
     function withdraw(address tokenType, uint256 amount, uint256 event_id, uint8[] memory v, bytes32[] memory r, bytes32[] memory s) payable external {
         require(withdrawalsActive, "withdrawals paused");
-        bytes32 message = abi.encodePacked(tokenType, amount, msg.sender);
-        bridge.call(bytes4(keccak256("verifyMessage(bytes,uint256,uint8[],bytes32[],bytes32[])")), message, event_id, v, r, s); 
-
+        bytes32 message = keccak256(abi.encodePacked(tokenType, amount, msg.sender));
```
Comment (jordy25519) [inline crml/erc20-peg/contracts/ERC20Peg.sol:46]:
```suggestion
        bytes32 message = keccak256(abi.encodePacked(tokenType, amount, msg.sender, event_id));
```

**Review by jordy25519 — APPROVED** [review]
lgtm, there are a few changes needed will make a PR

**Pull request closed without merging**